File: backend/pokerogue/src/main/java/com/pokerogue/helper/pokemon2/dto/PokemonAbilityResponse.java
package com.pokerogue.helper.pokemon2.dto;

import jakarta.annotation.Nullable;

@Nullable
public record PokemonAbilityResponse(String id, String name, String description, Boolean passive, Boolean hidden) {

    public static PokemonAbilityResponse from(String ability) {
        return new PokemonAbilityResponse("","","",false,false);
    }
}

File: android/remote/src/main/java/poke/rogue/helper/remote/dto/response/ability/PokemonAbilityResponse.kt
package poke.rogue.helper.remote.dto.response.ability

import kotlinx.serialization.Serializable

@Serializable
data class PokemonAbilityResponse(
    val id: String,
    val name: String,
    val description: String,
    val passive: Boolean,
    val hidden: Boolean,
)
